subprojects/launcher/src/integTest/groovy/org/gradle/launcher/CommandLineIntegrationSpec.groovy

```diff
@@ -239,4 +239,35 @@ class CommandLineIntegrationSpec extends AbstractIntegrationSpec {
         6                      | ['--parallel', '--max-workers=6', '-Dorg.gradle.workers.max=4']
         4                      | ['-Dorg.gradle.parallel=true', '--max-workers=4']
     }
+
+    @Unroll
+    def "Set log level using org.gradle.log.level"() {
+        def message = 'Expected message in the output'
+        buildFile << """
+            task assertLogging {
+                doLast {
+                    logger.${logLevel} '${message}'
+                    assert logger.${logLevel}Enabled
+                    if ('${nextLevel}') {
```

Comment:
I don't see the point test the next log level. It's basically testing that our logging works properly and is not related to providing the log level by system property.